app/uk/gov/hmrc/apiplatform/modules/uplift/services/UpliftService.scala

```diff
@@ -103,9 +103,9 @@ class UpliftService @Inject()(
       app <- applicationRepository.fetchVerifiableUpliftBy(verificationCode)
         .map(_.getOrElse(throw InvalidUpliftVerificationCode(verificationCode)))
 
-      result <- app.state.name match {
-        case State.PRODUCTION => verifyProduction(app)
-        case PENDING_REQUESTER_VERIFICATION => verifyPending(app)
+      result <- app match {
+        case a if a.isInPreProductionOrProduction => verifyProduction(app)
```

Comment:
Ah.  Should have mentioned - don't use these helper functions in matches as it prevents checks on complete matches.